engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/schema/DefaultAvroSchemaEvolution.scala

```diff
@@ -0,0 +1,106 @@
+package pl.touk.nussknacker.engine.avro.schema
+
+import java.io.{ByteArrayOutputStream, IOException}
+import java.nio.ByteBuffer
+
+import org.apache.avro.Schema
+import org.apache.avro.generic._
+import org.apache.avro.io.{DecoderFactory, EncoderFactory}
+
+/**
+  * It's base implementation of AvroSchemaEvolution. In this case strategy to evolve record to schema is as follows:
+  *
+  * serialize record to record schema -> deserialize record to provided schema (final schema)
+  *
+  * This strategy is based on Confluent implementation of: serialization and deserialization method. But we don't
+  * allocate bytes for MagicByte and Id, because we don't need it.
+  *
+  * For now it's easiest way to convert GenericContainer record to wanted schema.
+  */
+class DefaultAvroSchemaEvolution extends AvroSchemaEvolution with DatumReaderWriterMixin {
+
+  /**
+    * In future we can try to configure it
+    */
+  protected final  val useSchemaReflection = false
+
+  protected final val encoderFactory: EncoderFactory = EncoderFactory.get
+
+  protected final val decoderFactory = DecoderFactory.get
+
+  override def alignRecordToSchema(container: GenericContainer, schema: Schema): GenericContainer = {
+    val writerSchema = container.getSchema
+    if (writerSchema.equals(schema)) {
+      container
+    } else {
+      val serializedObject = serializeToRecordSchema(container, writerSchema)
+      val deserializedRecord = deserializePayloadToSchema(serializedObject, container, writerSchema, schema)
+      deserializedRecord.asInstanceOf[GenericContainer]
+    }
+  }
+
+  /**
+    * It's copy paste from AbstractKafkaAvroDeserializer#DeserializationContext.read with some modification.
+    * We pass there record buffer data and schema which will be used to convert record.
+    *
+    * @param payload
+    * @param record
+    * @param writerSchema
+    * @param readerSchema
+    * @return
+    */
+  protected def deserializePayloadToSchema(payload: Array[Byte], record: GenericContainer, writerSchema: Schema, readerSchema: Schema): Any = {
+    val reader = createDatumReader(record, writerSchema, readerSchema, useSchemaReflection)
+
+    try {
+      val buffer = ByteBuffer.wrap(payload)
+      val length = buffer.limit
+      if (writerSchema.getType == Schema.Type.BYTES) {
+        val bytes = new Array[Byte](length)
+        buffer.get(bytes, 0, length)
+        bytes
+      } else {
+        val start = buffer.position + buffer.arrayOffset
+        val binaryDecoder = decoderFactory.binaryDecoder(buffer.array, start, length, null)
+        val result = reader.read(record, binaryDecoder)
+        if (writerSchema.getType == Schema.Type.STRING) result.toString else result
+      }
+    } catch {
+      case exc@(_: RuntimeException | _: IOException) =>
```

Comment:
Why not to handle Exception instead?

Reply:
It was copied from confluent AbstractKafkaAvroSerializer ;) They catch only these two exceptions..